### PR TITLE
Modify Software Tests page of the developer's guide to be requirements-agnostic

### PR DESCRIPTION
**Pull request by Shiro-Raven**
**Context:**
Software Tests page was outdated and relied on listing the testing requirements instead of relying on the continuously updated `requirements-dev.txt` file. 

**Description of the Change:**
Instructions for installing the development requirements was added to the Installation page and the Requirements section of Software Tests has been modified to be more informative rather than instructive.

**Benefits:**
Less overhead to update the guide later if deps are added/removed.

[[sc-63429](https://app.shortcut.com/xanaduai/story/63429)]

### CODE REVIEW

File: doc/development/guide/installation.rst [hunk 1]
```diff
@@ -66,6 +66,14 @@ importing PennyLane in Python.
     requires ``pip install -e .`` to be re-run in the plugin repository
     for the changes to take effect.
 
+Apart from the core packages needed to run PennyLane. Some extra packages need
+to be installed for several development processes, such as linting, testing, and
+pre-commit quality checks to name a few. Those can be installed easily via ``pip``:
```
Comment (trbromley) [inline doc/development/guide/installation.rst:71]:
```suggestion
pre-commit quality checks. Those can be installed easily via ``pip``:
```

File: doc/development/guide/tests.rst [hunk 2]
```diff
@@ -3,17 +3,17 @@ Software tests
...
 Requirements
 ~~~~~~~~~~~~
-The PennyLane test suite requires the Python ``pytest`` package, as well as:
+The PennyLane test suite requires the Python ``pytest`` package, as well as
+some extentions thereof, for example:
 
 * ``pytest-cov``: determines test coverage
 * ``pytest-mock``: allows replacing components with dummy/mock objects
 * ``flaky``: manages tests with non-deterministic behaviour
+* ``pytest-benchmark``: benchmarks the performance of functions, and can be used to ensure consistent runtime.
```
Comment (trbromley) [inline doc/development/guide/tests.rst:12]:
```suggestion
* ``pytest-benchmark``: benchmarks the performance of functions, and can be used to ensure consistent runtime
```

File: doc/development/guide/tests.rst [hunk 3]
```diff
@@ -3,17 +3,17 @@ Software tests
...
+* ``pytest-xdist``: currently used to force some tests to run on the same thread to avoid race conditions
 
-These requirements can be installed via ``pip``:
-
-.. code-block:: bash
-
-    pip install pytest pytest-cov pytest-mock flaky
+If you properly followed the installation guide, you should have all of these packages and others installed in your
```
Comment (trbromley) [inline doc/development/guide/tests.rst:15]: We could provide a link to the installation guide.

**Review by trbromley — COMMENTED** [review]
Thanks @Shiro-Raven, just a few comments.

**Comment by Shiro-Raven** [comment]
Gonna close this PR due to a problem with sphinx not building the docs from the latest commit. Gonna create a new branch and PR.

**Pull request closed without merging**